spyglass/src/main/java/com/linkedin/android/spyglass/ui/MentionsEditText.java

```diff
@@ -341,48 +348,50 @@ private void pasteHoneycombImpl(@IntRange(from = 0) int min, @IntRange(from = 0)
             for (int i = 0; i < clip.getItemCount(); i++) {
                 ClipData.Item item = clip.getItemAt(i);
                 String selectedText = item.coerceToText(getContext()).toString();
+                MentionsEditable text = getMentionsText();
+                MentionSpan[] spans = text.getSpans(min, max, MentionSpan.class);
+                for (MentionSpan span : spans) {
+                    if (text.getSpanEnd(span) == min) {
+                        // Don't remove the span, which is not selected and min(i.e selection anchor) is at the end of that span.
```

Comment:
We need to remove the span between min and max. This is required because in SpannableStringBuilder#replace(int, int, CharSequence) existing spans within the Editable that entirely cover the replaced range are retained, but any that were strictly within the range that was replaced are removed. In our case the existing spans are retained if the selection entirely covers the span. So, we just remove the existing span and replace the new text with that span. (Please refer: span_retain_issue.gif)

We do not want to remove the span, when we want to paste anything just next to the existing span. In this case "text.getSpanEnd(span)" will be equal to min. (Please refer: span_remove_issue.gif)

![span_remove_issue](https://cloud.githubusercontent.com/assets/12628247/21853601/d299dff6-d83d-11e6-8904-ed7d013ab5c6.gif)
![span_retain_issue](https://cloud.githubusercontent.com/assets/12628247/21853602/d29df118-d83d-11e6-9d17-ee1b89781ceb.gif)